frontend/src/containers/pages/AuthoringTool/CanvasSideBar/ComponentProperties/ButtonPropertiesComponent.jsx

```diff
@@ -96,6 +154,109 @@ export default function ButtonPropertiesComponent({
           })}
         </Select>
       </FormControl>
+
+      <FormControl fullWidth className={styles.componentProperty}>
+        <div style={{ display: "flex" }}>
+          <CustomTextField
+            label="New Flag"
+            value={newFlag}
+            fullWidth
+            onChange={handleInputChange}
+          />
+          <Button onClick={handleAddFlag}>Add</Button>
+        </div>
+      </FormControl>
+
+      <FormControl fullWidth className={styles.componentProperty}>
+        <CustomInputLabel shrink className={styles.plusLabel}>
+          Select flags to be added
+        </CustomInputLabel>
+        <Select
+          multiple
+          className={styles.selectInput}
+          value={Object.keys(component.flagAdditions || {})}
+          // renderValue={(selected) =>
+          //   selected.filter((flag) => component.flagAdditions[flag]).length > 0
+          //     ? selected.filter((flag) => component.flagAdditions[flag]).join(", ")
+          //     : "Select flags to be added"
+          // }
+          renderValue={(selected) =>
+            selected.filter((flag) => component.flagAdditions[flag]).join(", ")
+          }
+          displayEmpty
+        >
+          {Object.keys(component.flagAdditions || {}).length > 0 ? (
+            <div style={{ display: "flex", flexDirection: "column" }}>
+              {Object.keys(component.flagAdditions).map((flag) => (
+                <FormControlLabel
+                  key={flag}
+                  control={
+                    <CustomCheckBox
+                      checked={component.flagAdditions[flag]}
+                      onChange={() => handleCheckboxChange(flag, true)}
+                    />
+                  }
+                  label={<span className={styles.plusLabel}>{flag}</span>}
+                />
+              ))}
+            </div>
+          ) : (
+            <Typography className={styles.menuItem}>
+              This button property currently does not have any flags. Please add
+              flags first using the text field above.
+            </Typography>
+          )}
+        </Select>
+      </FormControl>
+      <FormControl fullWidth className={styles.componentProperty}>
+        <CustomInputLabel shrink className={styles.minusLabel}>
+          Select flags to be removed
+        </CustomInputLabel>
+        <Select
+          // style={{ marginTop: "10px" }}
+          multiple
+          className={styles.selectInput}
+          value={Object.keys(component.flagDeletions || {})}
+          // renderValue={(selected) =>
+          //   selected.filter((flag) => component.flagDeletions[flag]).length > 0
+          //     ? selected.filter((flag) => component.flagDeletions[flag]).join(", ")
+          //     : "Select flags to be removed"
+          // }
```

Comment:
is this still needed?

Reply:
Oops, forgot to delete that part. I was experimenting with some things.